sdk/src/types/block/output/account.rs

```diff
@@ -564,56 +465,46 @@ impl AccountOutput {
             return Err(StateTransitionError::MutatedImmutableField);
         }
 
-        if next_state.state_index == current_state.state_index + 1 {
-            // State transition.
-            if current_state.state_controller_address() != next_state.state_controller_address()
-                || current_state.governor_address() != next_state.governor_address()
-                || current_state.features.metadata() != next_state.features.metadata()
-            {
-                return Err(StateTransitionError::MutatedFieldWithoutRights);
-            }
-
-            let created_foundries = outputs.iter().filter_map(|output| {
-                if let Output::Foundry(foundry) = output {
-                    if foundry.account_address().account_id() == &next_state.account_id
-                        && !input_chains.contains_key(&foundry.chain_id())
-                    {
-                        Some(foundry)
-                    } else {
-                        None
-                    }
+        // TODO update when TIP is updated
+        // // Governance transition.
+        // if current_state.amount != next_state.amount
+        //     || current_state.native_tokens != next_state.native_tokens
+        //     || current_state.foundry_counter != next_state.foundry_counter
+        // {
+        //     return Err(StateTransitionError::MutatedFieldWithoutRights);
+        // }
+
+        // // State transition.
+        // if current_state.features.metadata() != next_state.features.metadata() {
+        //     return Err(StateTransitionError::MutatedFieldWithoutRights);
+        // }
+
+        let created_foundries = outputs.iter().filter_map(|output| {
+            if let Output::Foundry(foundry) = output {
+                if foundry.account_address().account_id() == &next_state.account_id
+                    && !input_chains.contains_key(&foundry.chain_id())
+                {
+                    Some(foundry)
                 } else {
                     None
                 }
-            });
+            } else {
+                None
+            }
+        });
 
-            let mut created_foundries_count = 0;
+        let mut created_foundries_count = 0;
 
-            for foundry in created_foundries {
-                created_foundries_count += 1;
+        for foundry in created_foundries {
+            created_foundries_count += 1;
```

Comment:
seems like you could just use `iter().enumerate()` bc it's unconditionally incremented?

Reply:
wait ... you need it down below ... I see